android/expoview/src/main/java/host/exp/exponent/ExpoUpdatesAppLoader.java

```diff
@@ -0,0 +1,237 @@
+import expo.modules.updates.UpdatesUtils;
+import expo.modules.updates.db.DatabaseHolder;
+import expo.modules.updates.db.entity.UpdateEntity;
+import expo.modules.updates.launcher.Launcher;
+import expo.modules.updates.launcher.SelectionPolicy;
+import expo.modules.updates.launcher.SelectionPolicyNewest;
+import expo.modules.updates.loader.FileDownloader;
+import expo.modules.updates.loader.LoaderTask;
+import expo.modules.updates.manifest.Manifest;
+import host.exp.exponent.analytics.Analytics;
+import host.exp.exponent.analytics.EXL;
+import host.exp.exponent.di.NativeModuleDepsProvider;
+import host.exp.exponent.exceptions.ExceptionUtils;
+import host.exp.exponent.kernel.ExpoViewKernel;
+import host.exp.exponent.kernel.ExponentUrls;
+import host.exp.exponent.kernel.KernelConfig;
+import host.exp.exponent.storage.ExponentDB;
+import host.exp.exponent.storage.ExponentSharedPreferences;
+import host.exp.expoview.ExpoViewBuildConfig;
+import host.exp.expoview.Exponent;
+
+public abstract class ExpoUpdatesAppLoader {
+
+  @Inject
+  ExponentManifest mExponentManifest;
+
+  @Inject
+  ExponentSharedPreferences mExponentSharedPreferences;
+
+  @Inject
+  DatabaseHolder mDatabaseHolder;
+
+  private static final String TAG = ExpoUpdatesAppLoader.class.getSimpleName();
+
+  private String mManifestUrl;
+  private final boolean mUseCacheOnly;
+
+  private Context mContext;
+
+  public ExpoUpdatesAppLoader(String manifestUrl) {
+    this(manifestUrl, false, null);
+  }
+
+  public ExpoUpdatesAppLoader(String manifestUrl, boolean useCacheOnly, Context context) {
+    NativeModuleDepsProvider.getInstance().inject(ExpoUpdatesAppLoader.class, this);
+
+    mManifestUrl = manifestUrl;
+    mUseCacheOnly = useCacheOnly;
+    mContext = context;
+  }
+
+  public abstract void onOptimisticManifest(JSONObject optimisticManifest);
+
+  public abstract void onManifestCompleted(JSONObject manifest);
+
+  public abstract void onBundleCompleted(String localBundlePath);
+
+  public abstract void emitEvent(JSONObject params);
+
+  public abstract void onError(Exception e);
+
+  public abstract void onError(String e);
+
+  public void start() {
+    Uri manifestUrl = mExponentManifest.httpManifestUrl(mManifestUrl);
+
+    // TODO: handle development mode manifests
+
+    HashMap<String, Object> configMap = new HashMap<>();
+    configMap.put(UpdatesConfiguration.UPDATES_CONFIGURATION_UPDATE_URL_KEY, manifestUrl);
+    // TODO: if we want to use a scopeKey from the manifest here,
+    //  need to either keep track of URL -> scopeKey mapping separately
+    configMap.put(UpdatesConfiguration.UPDATES_CONFIGURATION_SCOPE_KEY_KEY, mManifestUrl);
+    configMap.put(UpdatesConfiguration.UPDATES_CONFIGURATION_SDK_VERSION_KEY, Constants.SDK_VERSIONS);
+    configMap.put(UpdatesConfiguration.UPDATES_CONFIGURATION_HAS_EMBEDDED_UPDATE, false);
+    configMap.put(UpdatesConfiguration.UPDATES_CONFIGURATION_ENABLED_KEY, Constants.ARE_REMOTE_UPDATES_ENABLED);
```

Comment:
Should updates always be enabled for the development client since we always need to load the in-development update? Or is this for managed standalone apps?

Reply:
Yes, this would be for managed standalone apps. This constant is set by the updates app.json configuration for standalone apps but is always `true` for the Expo client.